lib/attrsets.nix

```diff
@@ -125,6 +125,44 @@ rec {
   */
   concatMapAttrs = f: flip pipe [ (mapAttrs f) attrValues (foldl' mergeAttrs { }) ];
 
+  /*
+    `joinAttrsRecursive <joinPath> <upTo>` flattens nested attribute sets by
```

Comment:
```suggestion
    `flattenAttrsRecursive <joinPath> <upTo>` flattens nested attribute sets by
```

Reply:
It *flattens* the attributes but *joins* their names.

Reply:
Actually you can `join` `attrsOf`, based on Haskell's idea of what that word should mean.

```haskell
join :: Monad m => m (m a) -> m a
```

where `m` can be something like `attrsOf` (ignoring the fact that its `Monad` instance would be unlawful)

```haskell
join :: AttrsOf (AttrsOf a) -> AttrsOf a
```

Now make it recursive (not easily represented in Haskell syntax, fwiw), and additionally use the other meaning of `join` for the string combobulation and you have `joinAttrsRecursive`.

I think there's something poetic about using both meanings of "join", but maybe that's just me.

Anyway, `join` and `flatten` are close to synonymous between languages, and using a different word for the unlawful variation that we're dealing with here seems appropriate to me.

`flatten` may be easier to find though, so idk.

Reply:
Well, it's quite a stretch, but okay.